src/libsyntax/attr.rs

```diff
@@ -877,10 +886,38 @@ pub fn find_repr_attrs(diagnostic: &Handler, attr: &Attribute) -> Vec<ReprAttr>
                     if let Some(h) = hint {
                         acc.push(h);
                     }
+                } else if let Some((name, value)) = item.name_value_str() {
```

Comment:
My historical position was that `#[repr(align(16))]` is grammatically invalid (e.g. would simply not parse), but if we've changed the grammar since then then that seems fine by me.